pandas/core/indexes/category.py

```diff
@@ -769,5 +769,8 @@ def _wrap_joined_index(
         name = get_op_result_name(self, other)
         return self._create_from_codes(joined, name=name)
 
+    def replace(self, *args, **kwargs):
```

Comment:
can you make the signature match Index.replace anyway

possible follow-up: Categorical _does_ implement replace, so it should be easy to pass-through here

Reply:
Signature matches now.

The implementation of pd.Categorical isn't able to support the replace operation, since it defers part of the replace logic [here](https://github.com/pandas-dev/pandas/blob/master/pandas/core/arrays/categorical.py#L2249-L2250). I tried the `pd.CategoricalIndex().to_series()` route but that failed due to #37899.

I've seen a few more issues with Categorical related replaces, therefore suggest pushing this over the line and deferring `CategoricalIndex.replace()` to after those are fixed. If agreed, I'll create an issue for noting the lack of implementation here.